### ISSUE TITLE
error converting YAML to JSON: yaml: line 42: did not find expected '-' indicator

### ISSUE THREAD

**Issue by fsckzy** [hashicorp/consul-helm#449]
k8s 1.17.5
helm 3.2.1
Customizing  Installation error

```shell
[root@k8sm1 opt]# helm install consul hashicorp/consul -f config.yaml
Error: YAML parse error on consul/templates/server-statefulset.yaml: error converting YAML to JSON: yaml: line 42: did not find expected '-' indicato
```
config.yaml
```yaml
global:
  image: 'consul:1.5.0'
  domain: consul
  datacenter: primarydc
  bootstrapACLs: false
server:
  replicas: 3
  bootstrapExpect: 3
  resources: |
    requests:
      memory: "8Gi"
      cpu: "4"
      disk: "5Gi"
    limits:
      memory: "8Gi"
      cpu: "4"
      disk: "5Gi"
  affinity: |
    podAntiAffinity:
     requiredDuringSchedulingIgnoredDuringExecution:
       - labelSelector:
           matchLabels:
             app: {{ template "consul.name" . }}
             release: "{{ .Release.Name }}"
             component: server
       topologyKey: kubernetes.io/hostname
client:
  resources: |
    requests:
      memory: "2Gi"
      cpu: "2"
      disk: "5Gi"
    limits:
      memory: "2Gi"
      cpu: "2"
      disk: "5Gi"
ui:
  enabled: true
connectInject:
  enabled: true
  default: true
  aclBindingRuleSelector: “serviceaccount.name!=default”
```
please help me

**Comment by lkysow** [comment]
tl;dr, replace the quotes on that line with `"`, e.g. `"serviceaccount.name!=default"`


Hey, so first of all it says the error is on line 42 which is:
```
  aclBindingRuleSelector: “serviceaccount.name!=default”
```

If you look at the quotes there, notice how the quotes are slanted to the left: `”`. Now look at the quotes above it, e.g. in the disk setting: `"`. See how they're different? Your editor has inserted a different type of quote character for that setting.

To verify, you can use a tool like https://www.babelstone.co.uk/Unicode/whatisit.html:

![image](https://user-images.githubusercontent.com/1034429/81585525-2502a180-9369-11ea-9132-8f5fa1571cc0.png)

vs the other quotes:

![image](https://user-images.githubusercontent.com/1034429/81585578-351a8100-9369-11ea-82b7-bd446963524f.png)

So yeah, I'm not sure what editor you're using but code editors won't use that other quote character. Did you paste from our docs?

**Comment by drmp13** [comment]
I've got similar error:
Error: YAML parse error on consul/templates/server-statefulset.yaml: error converting YAML to JSON: yaml: line 43: did not find expected '-' indicator

I think it refer to file: https://github.com/hashicorp/consul-helm/blob/master/templates/server-statefulset.yaml

**Comment by lkysow** [comment]
@drmp13 can you provide us exact steps to reproduce please?

**Comment by jun-sheaf** [comment]
@lkysow Aside from the problem you mentioned, the `affinity` is not well-formed `yaml`. You may want to update the docs (https://learn.hashicorp.com/tutorials/consul/kubernetes-deployment-guide).
